src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Encapsulates a Parser, which parses user inputs, interprets them as commands and
+ * updates the TaskList.
+ *
+ * @author Hanif Kamal
+ */
+public class Parser {
+    /** The TaskList that the Parser updates after command interpretation */
+    private TaskList list;
+
+    /**
+     * Class constructor to initialize a Parser instance.
+     *
+     * @param list The TaskList that is to be updated.
+     */
+    public Parser(TaskList list) {
+        this.list = list;
+    }
+
+    /**
+     * Takes in a String which represents user input, and parses it. It executes the
+     * interpreted commands and then updates the TaskList accordingly.
+     *
+     * @param input The String that is input by the user.
+     * @throws DukeException In the case of invalid inputs.
+     */
+    public void parse(String input) throws DukeException {
+        String[] splitInput = input.split(" ", 2);
+        String first = splitInput[0];
```

Comment:
Should this be firstWord or firstParameter?

Reply:
Yes, it should! Thank you very much for spotting this error. I have endeavoured to fix my mistake.